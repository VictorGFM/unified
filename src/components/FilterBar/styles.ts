import styled from 'styled-components'
import { COLORS } from '../../styles/global'

export const FilterBarDiv = styled.div<{ isOpen: boolean }>`
  display: flex;
  flex-direction: column;

  width: 600px;
  height: 56px;
  margin-top: 15px;

  background-color: ${COLORS.WHITE.ONE};
  border: solid ${COLORS.GREY.THREE} 1px;
  border-radius: 7px;
  box-sizing: border-box;

  ${({ isOpen }) =>
    isOpen &&
    `
    height: 200px;
  `}
`

export const FilterBarHeader = styled.div`
  display: flex;
  align-items: center;

  margin-top: 7px;
`

export const SearchBarDiv = styled.div`
  position: relative;

  width: 530px;
  height: 34px;
  margin-left: 14px;
`

export const SearchBar = styled.input`
  width: 530px;
  height: 34px;
  padding-left: 9px;

  font-size: 17px;

  border: solid ${COLORS.GREY.ONE} 1px;
  border-radius: 5px;
  box-sizing: border-box;
`

export const SearchButton = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  top: 0px;
  right: 0px;

  height: 34px;
  margin-right: 7px;

  cursor: pointer;
`

export const SearchButtonIcon = styled.img`
  width: 20px;
  height: 20px;
`

export const FilterButton = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;

  width: 40px;
  height: 40px;
  margin-left: 7px;

  cursor: pointer;
  border-radius: 25px;

  &:hover {
    background-color: ${COLORS.PURPLE.TWO};
    img {
      filter: invert(44%) sepia(88%) saturate(6134%) hue-rotate(266deg) brightness(94%)
        contrast(97%);
    }
  }
`

export const FilterButtonIcon = styled.img`
  width: 30px;
  height: 30px;
`

export const FilterOptions = styled.div<{ isOpen: boolean }>`
  display: flex;
  justify-content: space-between;

  margin: 14px 63px 0px 23px;
  ${({ isOpen }) =>
    !isOpen &&
    `
    display: none;
  `}
`
