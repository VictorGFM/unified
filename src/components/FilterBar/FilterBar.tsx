import { FilterBarDiv } from './styles'

const FilterBar = () => {
  return <FilterBarDiv>TEST</FilterBarDiv>
}

export default FilterBar
