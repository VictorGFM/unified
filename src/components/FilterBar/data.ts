import { FiltersGroupsData } from './FilterBar'

export const Filters: FiltersGroupsData = {
  Posts: [
    {
      id: '1',
      label: 'Instagram Posts',
    },
    {
      id: '2',
      label: 'Twitter Posts',
    },
    {
      id: '3',
      label: 'Facebook Posts',
    },
  ],
  Groups: [
    {
      id: '4',
      label: 'Grupo dos Amigos',
    },
    {
      id: '5',
      label: 'Grupo do Trabalho',
    },
    {
      id: '6',
      label: 'Grupo da Família',
    },
    {
      id: '7',
      label: 'Grupo dos Canários',
    },
  ],
  Others: [
    {
      id: '8',
      label: 'Other Filter',
    },
    {
      id: '9',
      label: 'Other Filter',
    },
    {
      id: '10',
      label: 'Other Filter',
    },
  ],
}
