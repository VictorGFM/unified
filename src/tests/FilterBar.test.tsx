import { describe, expect, it } from 'vitest'
import { render } from '@testing-library/react'
import FilterBar from '../components/FilterBar'

describe('<FilterBar/>', () => {
  it('should show match snapshot', () => {
    const component = render(
      <FilterBar handleFilters={() => {}} handleSearchBar={() => {}} searchText={'some text'} />
    )

    expect(component).toMatchSnapshot()
  })
})
