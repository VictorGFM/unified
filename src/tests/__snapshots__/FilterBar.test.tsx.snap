// Vitest Snapshot v1

exports[`<FilterBar/> > should show match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ gvQIyN"
      >
        <div
          class="sc-gsnTZi izrTjg"
        >
          <div
            class="sc-crXcEl hPWHVC"
          >
            <input
              class="sc-evZas gYIiKi"
              placeholder="Search..."
            />
            <div
              class="sc-breuTD hJnmYB"
            >
              <img
                class="sc-ksZaOG crynEB"
                src="/src/assets/icons/SearchIcon.svg"
              />
            </div>
          </div>
          <div
            class="sc-dkzDqf kFWBN"
          >
            <img
              class="sc-hKMtZM iTFQlP"
              src="/src/assets/icons/FilterIcon.svg"
            />
          </div>
        </div>
        <div
          class="sc-eCYdqJ dryNwB"
        >
          <div
            class="sc-papXJ cklNIB"
          >
            <span
              class="sc-jqUVSM denPsi"
            >
              Posts
            </span>
            <hr
              class="sc-kDDrLX ifrIVl"
            />
            <div
              class="sc-iqcoie hkcSAG"
            >
              <div
                class="sc-jSMfEi dTXaiz"
              >
                <div
                  class="sc-gKXOVf knmpTR"
                >
                  <img
                    class="sc-iBkjds"
                    src="/src/assets/icons/CheckIcon.svg"
                  />
                </div>
                <span
                  class="sc-ftvSup khYnqs"
                >
                  Instagram Posts
                </span>
              </div>
              <div
                class="sc-jSMfEi dTXaiz"
              >
                <div
                  class="sc-gKXOVf knmpTR"
                >
                  <img
                    class="sc-iBkjds"
                    src="/src/assets/icons/CheckIcon.svg"
                  />
                </div>
                <span
                  class="sc-ftvSup khYnqs"
                >
                  Twitter Posts
                </span>
              </div>
              <div
                class="sc-jSMfEi dTXaiz"
              >
                <div
                  class="sc-gKXOVf knmpTR"
                >
                  <img
                    class="sc-iBkjds"
                    src="/src/assets/icons/CheckIcon.svg"
                  />
                </div>
                <span
                  class="sc-ftvSup khYnqs"
                >
                  Facebook Posts
                </span>
              </div>
            </div>
          </div>
          <div
            class="sc-papXJ cklNIB"
          >
            <span
              class="sc-jqUVSM denPsi"
            >
              Groups
            </span>
            <hr
              class="sc-kDDrLX ifrIVl"
            />
            <div
              class="sc-iqcoie hkcSAG"
            >
              <div
                class="sc-jSMfEi dTXaiz"
              >
                <div
                  class="sc-gKXOVf knmpTR"
                >
                  <img
                    class="sc-iBkjds"
                    src="/src/assets/icons/CheckIcon.svg"
                  />
                </div>
                <span
                  class="sc-ftvSup khYnqs"
                >
                  Grupo dos Amigos
                </span>
              </div>
              <div
                class="sc-jSMfEi dTXaiz"
              >
                <div
                  class="sc-gKXOVf knmpTR"
                >
                  <img
                    class="sc-iBkjds"
                    src="/src/assets/icons/CheckIcon.svg"
                  />
                </div>
                <span
                  class="sc-ftvSup khYnqs"
                >
                  Grupo do Trabalho
                </span>
              </div>
              <div
                class="sc-jSMfEi dTXaiz"
              >
                <div
                  class="sc-gKXOVf knmpTR"
                >
                  <img
                    class="sc-iBkjds"
                    src="/src/assets/icons/CheckIcon.svg"
                  />
                </div>
                <span
                  class="sc-ftvSup khYnqs"
                >
                  Grupo da Família
                </span>
              </div>
              <div
                class="sc-jSMfEi dTXaiz"
              >
                <div
                  class="sc-gKXOVf knmpTR"
                >
                  <img
                    class="sc-iBkjds"
                    src="/src/assets/icons/CheckIcon.svg"
                  />
                </div>
                <span
                  class="sc-ftvSup khYnqs"
                >
                  Grupo dos Canários
                </span>
              </div>
            </div>
          </div>
          <div
            class="sc-papXJ cklNIB"
          >
            <span
              class="sc-jqUVSM denPsi"
            >
              Others
            </span>
            <hr
              class="sc-kDDrLX ifrIVl"
            />
            <div
              class="sc-iqcoie hkcSAG"
            >
              <div
                class="sc-jSMfEi dTXaiz"
              >
                <div
                  class="sc-gKXOVf knmpTR"
                >
                  <img
                    class="sc-iBkjds"
                    src="/src/assets/icons/CheckIcon.svg"
                  />
                </div>
                <span
                  class="sc-ftvSup khYnqs"
                >
                  Other Filter
                </span>
              </div>
              <div
                class="sc-jSMfEi dTXaiz"
              >
                <div
                  class="sc-gKXOVf knmpTR"
                >
                  <img
                    class="sc-iBkjds"
                    src="/src/assets/icons/CheckIcon.svg"
                  />
                </div>
                <span
                  class="sc-ftvSup khYnqs"
                >
                  Other Filter
                </span>
              </div>
              <div
                class="sc-jSMfEi dTXaiz"
              >
                <div
                  class="sc-gKXOVf knmpTR"
                >
                  <img
                    class="sc-iBkjds"
                    src="/src/assets/icons/CheckIcon.svg"
                  />
                </div>
                <span
                  class="sc-ftvSup khYnqs"
                >
                  Other Filter
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ gvQIyN"
    >
      <div
        class="sc-gsnTZi izrTjg"
      >
        <div
          class="sc-crXcEl hPWHVC"
        >
          <input
            class="sc-evZas gYIiKi"
            placeholder="Search..."
          />
          <div
            class="sc-breuTD hJnmYB"
          >
            <img
              class="sc-ksZaOG crynEB"
              src="/src/assets/icons/SearchIcon.svg"
            />
          </div>
        </div>
        <div
          class="sc-dkzDqf kFWBN"
        >
          <img
            class="sc-hKMtZM iTFQlP"
            src="/src/assets/icons/FilterIcon.svg"
          />
        </div>
      </div>
      <div
        class="sc-eCYdqJ dryNwB"
      >
        <div
          class="sc-papXJ cklNIB"
        >
          <span
            class="sc-jqUVSM denPsi"
          >
            Posts
          </span>
          <hr
            class="sc-kDDrLX ifrIVl"
          />
          <div
            class="sc-iqcoie hkcSAG"
          >
            <div
              class="sc-jSMfEi dTXaiz"
            >
              <div
                class="sc-gKXOVf knmpTR"
              >
                <img
                  class="sc-iBkjds"
                  src="/src/assets/icons/CheckIcon.svg"
                />
              </div>
              <span
                class="sc-ftvSup khYnqs"
              >
                Instagram Posts
              </span>
            </div>
            <div
              class="sc-jSMfEi dTXaiz"
            >
              <div
                class="sc-gKXOVf knmpTR"
              >
                <img
                  class="sc-iBkjds"
                  src="/src/assets/icons/CheckIcon.svg"
                />
              </div>
              <span
                class="sc-ftvSup khYnqs"
              >
                Twitter Posts
              </span>
            </div>
            <div
              class="sc-jSMfEi dTXaiz"
            >
              <div
                class="sc-gKXOVf knmpTR"
              >
                <img
                  class="sc-iBkjds"
                  src="/src/assets/icons/CheckIcon.svg"
                />
              </div>
              <span
                class="sc-ftvSup khYnqs"
              >
                Facebook Posts
              </span>
            </div>
          </div>
        </div>
        <div
          class="sc-papXJ cklNIB"
        >
          <span
            class="sc-jqUVSM denPsi"
          >
            Groups
          </span>
          <hr
            class="sc-kDDrLX ifrIVl"
          />
          <div
            class="sc-iqcoie hkcSAG"
          >
            <div
              class="sc-jSMfEi dTXaiz"
            >
              <div
                class="sc-gKXOVf knmpTR"
              >
                <img
                  class="sc-iBkjds"
                  src="/src/assets/icons/CheckIcon.svg"
                />
              </div>
              <span
                class="sc-ftvSup khYnqs"
              >
                Grupo dos Amigos
              </span>
            </div>
            <div
              class="sc-jSMfEi dTXaiz"
            >
              <div
                class="sc-gKXOVf knmpTR"
              >
                <img
                  class="sc-iBkjds"
                  src="/src/assets/icons/CheckIcon.svg"
                />
              </div>
              <span
                class="sc-ftvSup khYnqs"
              >
                Grupo do Trabalho
              </span>
            </div>
            <div
              class="sc-jSMfEi dTXaiz"
            >
              <div
                class="sc-gKXOVf knmpTR"
              >
                <img
                  class="sc-iBkjds"
                  src="/src/assets/icons/CheckIcon.svg"
                />
              </div>
              <span
                class="sc-ftvSup khYnqs"
              >
                Grupo da Família
              </span>
            </div>
            <div
              class="sc-jSMfEi dTXaiz"
            >
              <div
                class="sc-gKXOVf knmpTR"
              >
                <img
                  class="sc-iBkjds"
                  src="/src/assets/icons/CheckIcon.svg"
                />
              </div>
              <span
                class="sc-ftvSup khYnqs"
              >
                Grupo dos Canários
              </span>
            </div>
          </div>
        </div>
        <div
          class="sc-papXJ cklNIB"
        >
          <span
            class="sc-jqUVSM denPsi"
          >
            Others
          </span>
          <hr
            class="sc-kDDrLX ifrIVl"
          />
          <div
            class="sc-iqcoie hkcSAG"
          >
            <div
              class="sc-jSMfEi dTXaiz"
            >
              <div
                class="sc-gKXOVf knmpTR"
              >
                <img
                  class="sc-iBkjds"
                  src="/src/assets/icons/CheckIcon.svg"
                />
              </div>
              <span
                class="sc-ftvSup khYnqs"
              >
                Other Filter
              </span>
            </div>
            <div
              class="sc-jSMfEi dTXaiz"
            >
              <div
                class="sc-gKXOVf knmpTR"
              >
                <img
                  class="sc-iBkjds"
                  src="/src/assets/icons/CheckIcon.svg"
                />
              </div>
              <span
                class="sc-ftvSup khYnqs"
              >
                Other Filter
              </span>
            </div>
            <div
              class="sc-jSMfEi dTXaiz"
            >
              <div
                class="sc-gKXOVf knmpTR"
              >
                <img
                  class="sc-iBkjds"
                  src="/src/assets/icons/CheckIcon.svg"
                />
              </div>
              <span
                class="sc-ftvSup khYnqs"
              >
                Other Filter
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
